mirrord/layer/src/socket/ops.rs

```diff
@@ -191,15 +201,26 @@ pub(super) fn bind(
         Err(HookError::AddressAlreadyBound(requested_address))?;
     }
 
-    let bind_result = unsafe { FN_BIND(sockfd, unbound_address.as_ptr(), unbound_address.len()) };
-    if bind_result != 0 {
-        error!(
-            "listen -> Failed `bind` sockfd {:#?} to address {:#?} with errno {:#?}!",
-            sockfd,
-            unbound_address,
-            errno::errno()
-        );
-        return Err(io::Error::last_os_error())?;
+    // Try to bind a port from listen ports, if no configuration
+    // try to bind the requested port, if not available get a random port
+    // if there's configuration and binding fails with the requested port
+    // we return address not available and not fallback to a random port.
+    let listen_port = LISTEN_PORTS
+        .get()
+        .expect("`LISTEN_PORTS` not set. Please report a bug")
+        .get_by_left(&requested_address.port())
+        .cloned();
+
+    // Listen port was specified
+    if let Some(port) = listen_port {
+        bind_port(sockfd, socket.domain, port)?;
+    } else {
+        bind_port(sockfd, socket.domain, requested_address.port()).or_else(|e| {
+            warn!(
+                "bind -> first `bind` failed on port {listen_port:?} with {e:?}, trying to bind to a random port"
+            );
+            bind_port(sockfd, socket.domain, 0)
+        })?
     }
```

Comment:
```suggestion
    if let Some(port) = listen_port {
        bind_port(sockfd, socket.domain, port)
    } else {
        bind_port(sockfd, socket.domain, requested_address.port()).or_else(|e| {
            warn!(
                "bind -> first `bind` failed on port {listen_port:?} with {e:?}, trying to bind to a random port"
            );
            bind_port(sockfd, socket.domain, 0)
        })
    }?;
```

I think this is valid Rust?